cmd/state-installer/installer.go

```diff
@@ -185,16 +185,14 @@ func detectCorruptedInstallDir(path string) error {
 	if err != nil {
 		return errs.Wrap(err, "Could not detect bin path")
 	}
-	if !fileutils.DirExists(binPath) {
```

Comment:
You'll have to account for this on line 203 where you try to read the `binPath`.

Reply:
Don't we want to fail if the `bin` path doesn't exist? With the check removed this will fail if the `binPath` does not exist

Reply:
ReadDir doesn't seems to return any identifiable error so I've added this check to the error case